### DISCUSSION TITLE
Zenoh-pico as an ESP-IDF component?

### DISCUSSION THREAD

**Discussion by nagygabor93** [eclipse-zenoh/roadmap#85] [Q&A] [answered]
Hi!
I would like to use zenoh (and zenoh pico) for one of our projects. Our ESP32 firmware uses the "native" ESP-IDF dev environment . Zenoh Pico suggest using PlatformIO but we don't want to introduce yet again a new system... 

Is it possible to build zenoh-pico as a simple ESP-IDF component, by just using the provided cmake file?

**Comment by cguimaraes** [comment, marked as answer]
That is one of the goals for the long-term for Zenoh-Pico...to provide native support for each platform: https://github.com/eclipse-zenoh/zenoh-pico/issues/93

However, there are more pressing features that we are focusing on the short- and mid-term roadmap.
If you have experience with ESP-IDF, we would really appreciate and gladly accept any PR you could provide.

**Reply by nagygabor93** [reply]
Thanks for the answer! 

I have experience with ESP-IDF, but on the other hand, not a deep knowledge on zenoh-pico (yet) :D 

I can not find info for example, what the purpose of the extra_script.py in the zenoh-pico repo etc.

Also I think the recommended installation in the zenoh-pico repository for ESP-IDF is misleading, since it is a complete copy paste example from the Arduino one. 

I briefly checked the src files and it seems to me that there is not any magic there (like generating code from scripts or anything like that) so maybe the best course of action for me is to create a new cmake file to be able to include it to an ESP-IDF project as a component.

**Reply by cguimaraes** [reply]
The `extra_script.py` in zenoh-pico is regarding PlatformIO.
And indeed, with PlatformIO the struct of the projects are pretty much the same. However, the example is different: https://github.com/eclipse-zenoh/zenoh-pico/tree/master/examples/espidf

We can provide you more close support and discussion on our Discord channel.
Feel free to join and ping me there.

**Comment by nagygabor93** [comment]
If someone interested, I have ported the zenoh-pico to work in the esp-idf environment as a component. Nothing fancy, but it works:
https://github.com/nagygabor93/zenoh-pico

**Reply by cguimaraes** [reply]
Feel free to contribute it back with your own PR on Zenoh-Pico.
